File: src/main/java/com/matera/restserver/RestServerApplication.java
package com.matera.restserver;

import java.io.IOException;
import java.io.InputStream;

import java.util.List;
import java.util.logging.Logger;

import org.springframework.boot.CommandLineRunner;
import org.springframework.boot.SpringApplication;
import org.springframework.boot.autoconfigure.SpringBootApplication;
import org.springframework.context.annotation.Bean;

import com.fasterxml.jackson.core.type.TypeReference;
import com.fasterxml.jackson.databind.ObjectMapper;
import com.matera.restserver.service.EmployeeService;
import com.matera.restserver.util.Messages;
import com.matera.restserver.dto.EmployeeDTO;

@SpringBootApplication
public class RestServerApplication {

	private static final Logger LOGGER = Logger.getLogger( RestServerApplication.class.getName() );
	
	public static void main(String[] args) {
		SpringApplication.run(RestServerApplication.class, args);
	}

	@Bean
	CommandLineRunner runner(EmployeeService service) {
		return args -> {
			ObjectMapper mapper = new ObjectMapper();
			TypeReference<List<EmployeeDTO>> typeReference = new TypeReference<List<EmployeeDTO>>(){};
			InputStream inputStream = TypeReference.class.getResourceAsStream("/employees.json");
			try {
				List<EmployeeDTO> employees = mapper.readValue(inputStream,typeReference);
				service.create(employees);
				LOGGER.info(Messages.getMessage("employeescharge.success"));
			} catch (IOException e){
				LOGGER.severe(Messages.getMessage("employeescharge.fail") + e.getMessage());
			}
		};
	}
	
}

File: src/main/kotlin/com/matera/restserver/RestServerApplication.kt
package com.matera.restserver

import org.springframework.boot.autoconfigure.SpringBootApplication
import org.springframework.boot.autoconfigure.domain.EntityScan
import org.springframework.boot.context.properties.ConfigurationPropertiesScan
import org.springframework.boot.runApplication
import org.springframework.context.annotation.ComponentScan
import org.springframework.data.jpa.repository.config.EnableJpaRepositories

@ComponentScan("com.matera.restserver")
@EnableJpaRepositories("com.matera.restserver.repository")
@EntityScan("com.matera.restserver.model")
@ConfigurationPropertiesScan
@SpringBootApplication
open class RestServerApplication
    fun main(args: Array<String>) {
        runApplication<RestServerApplication>(*args)
    }